src/analyses/ai.cpp

```diff
@@ -197,16 +198,18 @@ void ai_baset::finalize()
   // Nothing to do per default
 }
 
-ai_baset::locationt ai_baset::get_next(
-  working_sett &working_set)
+ai_baset::trace_ptrt ai_baset::get_next(working_sett &working_set)
 {
   PRECONDITION(!working_set.empty());
 
-  working_sett::iterator i=working_set.begin();
-  locationt l=i->second;
-  working_set.erase(i);
+  // STL guarantees this will be key minimal
```

Comment:
If you're relying on this you should probably static-assert that `working_sett` is the type you expect

Reply:
Done.